tensorflow/python/keras/_impl/keras/callbacks_test.py

```diff
@@ -498,7 +501,10 @@ def data_generator():
       values = []
       with open(fp) as f:
         for x in csv.reader(f):
-          values.append(x)
+          # In windows, due to \n\r line ends we may end up reading empty lines
```

Comment:
Nit: s/\n\r/\r\n/?

...also I'm surprised that `csv.reader()` isn't robust to this. Perhaps [this Stack Overflow answer](https://stackoverflow.com/a/18899548/3574081), which suggests using `with open(fp, 'rU') as f:`, would avoid the need for this workaround?

Reply:
Done

Reply:
This did not work, so I have reverted to checking if the line is empty and skipping them.